### DISCUSSION TITLE
Lots of messy changes / ideas: Landscape, Samsung s6 lite formatted planner, ADHD-Planner, Moodtracker, and other...  quite some messy "work"

### DISCUSSION THREAD

**Discussion by tschnibo** [kudrykv/latex-yearly-planner#33] [Ideas]
**Hey @kudrykv, thank you a lot for your work!!! I use your planner extensively.** :1st_place_medal: 

But also, I tried to adapt it to the slightly different screen-format of the samsung s6 lite. **I wanted to use it in landscape.**

Well, I am really not proud of me, because:
- I invested a lot of time for "messing" with your neat work
  - e.g. I started to replace the minipages with paracol
  - i introduced quite some new concepts
- I have not really shared anything until now (mostly because I don't see my work as "quality", but of course I am getting somewhere still)
- I know this all is silly

When I see ideas like the Kanban-Boards. I think on one hand, "I could mabe even implement that, at the moment". But also, I have messed so much with the basis, that I would need to spend quite a lot of time to merge my mess into the supernote / remarkable formats.

**So for the moment:**
- If someone has a **galaxy s6 lite**, I could maybe provide an adapted landscape planner for it ;) 
- for some "inspiration", I happily share my messy planner, as it is at the moment. I had to split the year into 4 quarters, just because I included too many extra note pages... as you might gather, this is quite a specific ADHD-related implementation.

Just trying to partially fix my silliness by open communication. In order to publish, or even make "pull requests" I would need to do a lot of work still.

@kudrykv (and others). If you are interested, I would be happy to chat about maybe some ways I could actually support the project.

Quarter3 File attached. 
[custom.wide.base.border.dotted.default.dailycal.2022.3.pdf](https://github.com/kudrykv/latex-yearly-planner/files/9078929/custom.wide.base.border.dotted.default.dailycal.2022.3.pdf)

One can easily see that this is still "work in progress", I have used some design-variants, just to figure out which elements I like the most. Not decided yet. I'll just start with the state as it is, for the next 2.5 months.

As this was requested too: **activity tracker**, some simple implementation as it just was crafted tonight: 
![grafik](https://user-images.githubusercontent.com/16188592/178142720-60205eb7-8853-4592-8a62-4a467154936a.png)

( ... or just see the pdf...)

Best regards, tschnibo

edit: just included minor cosmetic changes

**Comment by kudrykv** [comment]
Hi @tschnibo!

Thank you for being interested in this project!
The landscape version is something that it doesn't have and could be something this project would benefit from.

> I am really not proud of me

Well, you should be 🙂 You were able to weed through the code and do updates. I, for one, abandoned that in favor of a refactor (hehe).
It's not silly. You're getting familiar with Go code and with TeX. (The question, though, is whether you'll need it somewhere else besides this project)

It would be great to have contributions and extensions. Unfortunately, I'm a blocker at the moment, as I'm in the middle of yet another refactoring. I'm hoping that it would stay, however. Its idea is to make it easier to extend existing planners. Building new ones would still be a challenge as, well, new planners need to be built from the ground up.

You can check out `refactor` branch. It's still a hot mess and I'll likely change the structure of lower components a bunch of times, but the general view and idea should be comprehensible (check the `app2` folder).

To sum up: I would like others to contribute their visions and ideas; it's me who needs to unblock that first.

**Reply by tschnibo** [reply]
Hey @kudrykv,

thank you for your quick response! And your kind words!

Well, actually I have never touched Golang before (rather elixir, python...) and used latex mostly for University... I don't think that I am ever going to "master" latex. ;) And with go, this is not even a goal for me ;).

I'll have a look at your refactor branch. Could you give me a hint, what are the major changes? (philosophy, packages?)
I was struggling with the minipages approach, and therefore switched to **paracol** for the new pages, and I used **tcolorbox** extensively.

In general, I was hoping to set up a more flexible pipeline-logic, where the single elements are more easily plugged in - better customization.
But I was struggling a lot, in many ways. But who isn't with latex I guess ;)

I think it would be nice to have a general notion of a "Grid", and the different screen formats could be adressed more flexible. The components _should_ be floating as much as possible. But I don't think that I have a working principle for these ideas. I just used one screen size ;).

It is my understanding, that only adobe pdf readers support "add page from template, via button and javascript inside pdf-form", otherwise this would be very nice. At least on android I have not found this.

Absolute "future goal" for me (completely out of scope), would be a web-based tool, which supports inking, and more dynamic component handling. Mixture between drawingboard and personal-wiki/personal-diary. This would enable proper analysis from things like mood-trackers and such. As I am "ok" with my planner for the moment (up to minor changes), I'll try to not put much more effort in this project., at least for the next weeks (Q3 ;) ), afterwards, I have not really decided yet, depends on my further needs, and also on the progress here :). It is good to see that you are still working on it, because as I said. I really like your work :1st_place_medal: 

Thank you a lot! :) And just let me know, in case I can support somehow!

Wishing you a nice weekend!
Best, Tschnibo

edit: in case you are interested, I can push my changes to a fork... (takes me maybe some time to clean it a bit beforehand, but well, messy it is going to be anyways ;) )

**Reply by kudrykv** [reply]
The idea of the refactor is to streamline the creation of a planner as much as possible. Therefore, it should be a simple, clean code that describes what happens at all times.

> In general, I was hoping to set up a more flexible pipeline logic, where the single elements are more easily plugged in - better customization.

That may be the seventh refactor, haha. Latex isn't a webpage, and it doesn't allow placing content arbitrarily on the page. So maybe, latex wasn't the best choice after all. It still suffices the needs, though—its typesetting is helpful in many ways.

Regards of "build-all tool" that would build _any_ planner, so to speak, it could be a blessing and a curse: if you have a few planners to choose from, you choose one. [If you have thousands, you don't pick any](https://medium.com/@FlorentGeerts/the-jam-experiment-how-choice-overloads-makes-consumers-buy-less-d610f8c37b9b).

So, let's do our best and avoid feature creep 🙂

**Reply by jstvz** [reply]
Howdy @tschnibo! I would be interested in seeing your changes (especially your activity tracker). I'm planning on something like it, and seeing what you did (no matter how messy 😉) would be helpful and probably save me some time.

(Thanks to @kudrykv for this project. A cleaner and hackable version of the calendars Sony put out for the DPT. Love it.)
